Platforms/QemuSbsaPkg/QemuVideoDxe/Qemu.h

```diff
@@ -0,0 +1,504 @@
+                                Language from the point of view of the driver
+                                specified by This.
+
+  @retval EFI_SUCCESS           The Unicode string for the user readable name in
+                                the language specified by Language for the
+                                driver specified by This was returned in
+                                DriverName.
+
+  @retval EFI_INVALID_PARAMETER ControllerHandle is not a valid EFI_HANDLE.
+
+  @retval EFI_INVALID_PARAMETER ChildHandle is not NULL and it is not a valid
+                                EFI_HANDLE.
+
+  @retval EFI_INVALID_PARAMETER Language is NULL.
+
+  @retval EFI_INVALID_PARAMETER ControllerName is NULL.
+
+  @retval EFI_UNSUPPORTED       The driver specified by This is not currently
+                                managing the controller specified by
+                                ControllerHandle and ChildHandle.
+
+  @retval EFI_UNSUPPORTED       The driver specified by This does not support
+                                the language specified by Language.
+
+**/
+EFI_STATUS
+EFIAPI
+QemuVideoComponentNameGetControllerName (
+  IN  EFI_COMPONENT_NAME_PROTOCOL  *This,
+  IN  EFI_HANDLE                   ControllerHandle,
+  IN  EFI_HANDLE                   ChildHandle        OPTIONAL,
+  IN  CHAR8                        *Language,
+  OUT CHAR16                       **ControllerName
+  );
+
+//
+// Local Function Prototypes
+//
+VOID
+InitializeCirrusGraphicsMode (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  QEMU_VIDEO_CIRRUS_MODES  *ModeData
+  );
+
+VOID
+InitializeBochsGraphicsMode (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  QEMU_VIDEO_MODE_DATA     *ModeData
+  );
+
+VOID
+SetPaletteColor (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  UINTN                    Index,
+  UINT8                    Red,
+  UINT8                    Green,
+  UINT8                    Blue
+  );
+
+VOID
+SetDefaultPalette (
+  QEMU_VIDEO_PRIVATE_DATA  *Private
+  );
+
+VOID
+DrawLogo (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  UINTN                    ScreenWidth,
+  UINTN                    ScreenHeight
+  );
+
+VOID
+outb (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  UINTN                    Address,
+  UINT8                    Data
+  );
+
+VOID
+outw (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  UINTN                    Address,
+  UINT16                   Data
+  );
+
+UINT8
+inb (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  UINTN                    Address
+  );
+
+UINT16
+inw (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  UINTN                    Address
+  );
+
+VOID
+BochsWrite (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  UINT16                   Reg,
+  UINT16                   Data
+  );
+
+UINT16
+BochsRead (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  UINT16                   Reg
+  );
+
+VOID
+VgaOutb (
+  QEMU_VIDEO_PRIVATE_DATA  *Private,
+  UINTN                    Reg,
+  UINT8                    Data
+  );
+
+EFI_STATUS
+QemuVideoCirrusModeSetup (

```

Comment:
Are we using this for SBSA platform?

Reply:
I don't think so. I'll strip out the Cirrus stuff.